rfc/004-abci++.md

```diff
@@ -0,0 +1,232 @@
+# RFC 004: ABCI++
+
+## Changelog
+
+- January 11, 2020: initialized
+
+## Author(s)
+
+- Dev (@valardragon)
+- Sunny (@sunnya97)
+
+## Context
```

Comment:
The context should make note of what the new (renamed?) `Finalize Block` step indicates. It should probably also make note of what happens to the old phases (prevote & precommit).

Reply:
Prevote and precommit aren't phases over ABCI. (ABCI doesn't know anything about them)

I'll add comments for Finalize block!